staging/src/k8s.io/cloud-provider/cloud.go

```diff
@@ -108,6 +108,9 @@ func GetInstanceProviderID(ctx context.Context, cloud Interface, nodeName types.
 		if err == NotImplemented {
 			return "", err
 		}
+		if err == InstanceNotFound {
+			return "", err
+		}
 
 		return "", fmt.Errorf("failed to get instance ID from cloud provider: %v", err)
```

Comment:
Since we are doing this it seems more appropiate to wrap the error so we can not keep growing the if clauses
```suggestion
		return "", fmt.Errorf("failed to get instance ID from cloud provider: %w", err)
```

Reply:
Sure, no problem